code/modules/mob/living/simple_animal/friendly/cat.dm

```diff
@@ -204,6 +204,30 @@
 		if("harm")
 			wuv(-1, M)
 
+/mob/living/simple_animal/pet/cat/attack_alien(mob/living/carbon/alien/humanoid/M)	//what the xeno was after all along in Alien
+	if(M.a_intent == "help" && M.caste == "h")
+		if (icon_state == "cat" || icon_state == "cat_rest" || icon_state == "cat_sit")
+			visible_message("<span class='name'>[M]</span> picks up and hugs <span class='name'>[src]</span>.")
+			src.Move(M)

```

Comment:
Calling a proc without a target beforehand (in this case, `src.`) calls it on the object you're in by default. Having a `src.` does the same thing as not, but `src.` is slower.